src/app/loading.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Spinner = () => {
+    return (
+        <div className="flex items-center justify-center h-screen">
+            <div className="animate-spin rounded-full h-5 w-5 border-t-2 border-b-2 border-gray-900"></div>
+        </div>
+    );
+};
```

Comment:
Would it be better to extract this into a common component and use here instead?

The loading in nextjs has it's own specific use case. More about it. https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming

Reply:
This can be done . Thanks, will fix.